shell/common/node_bindings.cc

```diff
@@ -689,6 +745,35 @@ void NodeBindings::StartPolling() {
   UvRunOnce();
 }
 
+void NodeBindings::SetAppCodeLoaded() {
+  app_code_loaded_ = true;
+}
+
+void NodeBindings::JoinAppCode() {
+  // We can only "join" app code to the main thread in the browser process
+  if (browser_env_ != BrowserEnvironment::kBrowser) {
+    return;
+  }
+
+  auto* browser = Browser::Get();
+  node::Environment* env = uv_env();
+
+  if (!env)
+    return;
+
+  v8::HandleScope handle_scope(env->isolate());
+  // Enter node context while dealing with uv events.
+  v8::Context::Scope context_scope(env->context());
+
+  // Pump the event loop until we get the signal that the app code has finished
+  // loading
+  while (!app_code_loaded_ && !browser->is_shutting_down()) {
+    int r = uv_run(uv_loop_, UV_RUN_NOWAIT);
+    if (r == 0)
+      base::RunLoop().QuitWhenIdle();  // Quit from uv.
```

Comment:
do we need to `break` or something here? seems wrong to call `uv_run` again after this